tasks.yaml

```diff
@@ -0,0 +1,81 @@
+tasks:
+  - name: set-modules-variable
+    actions:
+      - cmd: echo $(find . -type f -name 'go.mod' -exec dirname {} \; | cut -c 3-)
+        setVariables:
+          - name: MODULES
+
+  # This could be improved if something like this was introduced https://github.com/defenseunicorns/maru-runner/issues/17
+  - name: build-pkg
+    description: build specified package
+    actions:
+      - cmd: cd "${PKG}" &&  go build .
+
+  # Including a build step for codeQL
+  - name: build-all
+    description: build all packages
+    actions:
+      - task: set-modules-variable
+      - cmd: |
+          for dir in ${MODULES}; do
+            uds run build-pkg --set PKG=$dir
+          done
+
+  - name: vet-pkg
+    description: vet specified package
+    actions:
+      - cmd: cd "${PKG}" &&  go vet .
+
+  - name: vet-all
+    description: vet all packages
+    actions:
+      - task: set-modules-variable
+      - cmd: |
+          for dir in ${MODULES}; do
+            uds run vet-pkg --set PKG=$dir
+          done
+
+  - name: gosec-pkg
+    description: run go sec on specified package
+    actions:
+      - cmd: cd "${PKG}" &&  gosec . -no-fail
+
+  - name: gosec-all
+    description: gosec all packages
+    actions:
+      - task: set-modules-variable
+      - cmd: |
+          for dir in ${MODULES}; do
+            uds run gosec-pkg --set PKG=$dir
+          done
+
+  - name: test-pkg
+    description: Run unit tests for one package
+    actions:
+      - cmd: cd "${PKG}" && go test ./...
+
+  - name: test-all
+    description: Run unit tests in all packages
+    actions:
+      - task: set-modules-variable
+      - cmd: |
+          for dir in ${MODULES}; do
+            uds run test-pkg --set PKG=$dir
+          done
+
+  - name: lint-pkg
+    description: lint package
+    actions:
+      - cmd: cd "${PKG}" && revive -config ../revive.toml ./...
+
+  - name: lint-all
+    description: lint all packages
+    actions:
+      - cmd: revive ./...
+
+  - name: scan-cves
+    description: lint all packages
+    actions:
+      - task: set-modules-variable
+      - cmd: |
+          syft scan . -o json | grype --fail-on low
```

Comment:
I would put a TODO / note here to keep an eye on `.grype.yaml` in `zarf` to ensure we have parity across ignored CVEs

Reply:
IMO we should keep them separate, we probably won't have a lot of CVEs in common and even then they could be a false positive in zarf but a real CVE in pkg